src/edu/ucsb/nceas/metacat/dataone/D1AuthHelper.java

```diff
@@ -726,54 +715,72 @@ public boolean isLocalCNAdmin(Session session) throws ServiceFailure {
 /////////////  /////////////  
 //implementations         //
 /////////////  /////////////  
-       
-       
+
+
     /**
-     * Checks Metacat properties representing the local Node document for
-     * matching Node.subjects.  The NodeType parameter can be set to limit this 
-     * authorization check if needed.
+     * Checks Metacat properties representing the local Node document for matching Node.subjects.
+     * The NodeType parameter can be set to limit this authorization check if needed.
+     *
      * @param session
-     * @param nodeType  
+     * @param nodeType
      * @return
      * @throws ServiceFailure
      */
     protected boolean isLocalNodeAdmin(Session session, NodeType nodeType) throws ServiceFailure {
-
         boolean allowed = false;
```

Comment:
I find it harder to follow the logic of methods like this, that initialize a boolean, then assign different values to it in different blocks, and finally return it at the end. I think it's clearer and easier to follow, if the return value is simply returned immediately instead of assigned for returning later, so you can tell by looking at the code, exactly what the return value is. I have annotated the relevant lines in the code below, to show what I mean:

Reply:
first, delete line 730; then...

Reply:
Thank you for outlining how to refactor/improve this method. I have made the changes, and it is indeed a lot easier to follow/understand now.